src/component_category.rs

```diff
@@ -0,0 +1,121 @@
+// License: MIT
+// Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+//! This module defines the `ComponentCategory` enum, which represents the
+//! category of a component.
+
+use crate::graph_traits::Node;
+use std::fmt::Display;
+
+/// Represents the type of an inverter.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum InverterType {
+    Unspecified,
+    Solar,
+    Battery,
+    Hybrid,
+}
+
+impl Display for InverterType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            InverterType::Unspecified => write!(f, "Unspecified"),
+            InverterType::Solar => write!(f, "Solar"),
+            InverterType::Battery => write!(f, "Battery"),
+            InverterType::Hybrid => write!(f, "Hybrid"),
+        }
+    }
+}
+
+/// Represents the category of a component.
+///
+/// Values of the underlying generated `ComponentCategory` and `ComponentType` types
+/// need to be converted to this type, so that they can be used in the
+/// `ComponentGraph`.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ComponentCategory {
+    Unspecified,
+    Grid,
+    Meter,
+    Battery,
```

Comment:
We also have `BatteryType` (in the new microgrid API version), I would do the same for this type, even when in principle it doesn't seem to be as useful as with inverters, for the sake of consistency and in case it becomes more important in the future.

Reply:
done.